src/motion.rs

```diff
@@ -0,0 +1,208 @@
+#![deny(missing_docs)]
+//! Motion.
+
+use std::{fs, path::Path};
+
+use crate::{
+    core::Model,
+    error::CubismResult,
+    json::motion::{Motion3, Segment, SegmentPoint},
+};
+
+trait Interpolate {
```

Comment:
Do we need a trait for this? I assume you got something on your mind regarding this where it will be necessary in the future?

Reply:
In Cubism framework, a interpolation function is assigned to each segment as a function pointer but it is used nowhere but in `CubismMotion.cpp`, so a trait is too much for this. A subroutine is enough.

```rust
fn segment_intersects(seg: &Segment, t: f32) -> bool {
    match seg {
...
```

Fixed in 177e920.